refactor(app): display progress for active runs and allow log download on completion

rclarke0 opened the pull request:
# Overview

Displays run progress steps only when run is active and shows download run log button upon completion.


## Test Plan and Hands on Testing

Smoke tested
<img width="1008" height="336" alt="Screenshot 2025-11-10 at 10 12 34 AM" src="https://github.com/user-attachments/assets/095dd596-98b7-486e-b767-fd23f6e607ef" />
<img width="897" height="239" alt="Screenshot 2025-11-10 at 10 13 37 AM" src="https://github.com/user-attachments/assets/3ab8346e-0f52-4732-a0b0-1db92bb019c4" />
<img width="898" height="200" alt="Screenshot 2025-11-10 at 10 13 52 AM" src="https://github.com/user-attachments/assets/f87fff94-c563-48dc-9433-afcb6a1a9c0a" />



## Changelog

List changes introduced by this PR considering future developers and the end user. Give careful thought and clear documentation to breaking changes.


## Risk assessment

low

Closes EXEC-2043

koji committed:
fix(protocol-designer) avoid the white screen by removing liquid (#20072)

CaseyBatten committed:
fix(api): Ensure live stream camera default migrates to new default (#20077)

Ensures the configuration file that use to default to "NONE" now reinforces the new oe-core default.

ecormany committed:
chore(protocol-designer): release note addition for liquid deletion bug fix (#20076)

# Overview

Adding a bullet to release notes for #20072.

## Changelog

- One new bullet in 8.6.3 section.
- Remove some duplicate lines further down the notes file.

## Review requests

words good?

## Risk assessment

v low

ryanthecoder committed:
fix(api): add checking for imposible dynamic moves (#20054)

# Overview
This PR updates the OT3 Controller's motion planning to adjust the
commanded speed for a multi axis move if one of the plunger's are in the
moving axis. this ensures that the pipettes' plunger will move at the
speed commanded.

Right now we only move the plunger and other axis during dynamic
pipetting so this won't affect anything else in the system.

After doing it's best to adjust this speed, if the move is still
impossible an error will be logged into the api logs.

mjhuff committed:
refactor(app): Cherry-pick support notifications in secondary windows (#20040) (#20080)

rclarke0 committed:
fix(app): correcting text, wrapping module icons, and correcting text color on camera settings (#20061)

# Overview

Removed wrapping behavior of module icons in protocol card and robot
card
Corrected text on camera settings in run setup
Updated camera enabled text to a chip

## Test Plan and Hands on Testing

Smoke tested: 
<img width="931" height="332" alt="Screenshot 2025-11-06 at 12 35 10 PM"
src="https://github.com/user-attachments/assets/696b8acd-a14f-4b32-95c6-7643240d3ee5"
/>
<img width="931" height="149" alt="Screenshot 2025-11-06 at 4 11 08 PM"
src="https://github.com/user-attachments/assets/b37ebf6b-7a86-4b6b-95f8-98950bbbe47e"
/>


<img width="900" height="547" alt="Screenshot 2025-11-06 at 3 15 43 PM"
src="https://github.com/user-attachments/assets/d8198898-5852-4930-9830-85385138b68c"
/>

<img width="913" height="564" alt="Screenshot 2025-11-06 at 3 21 59 PM"
src="https://github.com/user-attachments/assets/7a5e8bcd-6cab-417a-9724-ac342613d165"
/>
<img width="991" height="617" alt="Screenshot 2025-11-06 at 4 19 05 PM"
src="https://github.com/user-attachments/assets/d4c4266a-45b0-4c3d-874b-5078a79c564c"
/>



## Risk assessment
low

Closes RQA-4819
Closes RQA-4825
Closes RQA-4824
Closes RQA-4830
Closes RQA-4823
Closes RQA-4835

mjhuff committed:
fix(api): correct image filenames (#20090)

rclarke0 committed:
refactor(app): runProgressMeter updates

step count only visible when run status is running and download run log only available when run is finished

rclarke0 committed:
refactor(app): runProgressMeter updates

step count only visible when run status is running and download run log only available when run is finished

mjhuff committed:
fix(app, app-shell): Refresh secondary window stream on new run (#20087)

Currently, we inject the `runId` into the secondary window, and that
`runId` serves as the basis for determining whether or not we should
render the stream. However, if a user keeps the secondary window open
when creating a new run on the same robot, the stream still reflects the
old run.

To fix, we check if this robot has a current run, and use that as the
basis for rendering the livestream. All of this works as expected
largely because we block the "show livestream" button for historical
runs.

koji committed:
chore: update vite version for patches (#20085)

* chore: update vite version for patches

ncdiehl11 committed:
style(step-generation): remove console log (#20092)

remove unnecessary console log in heater shaker command creator in step gen

rclarke0 committed:
fix(app): Conditionally change check preferences camera text color  (#20082)

# Overview

Changes the `check_preferences` text to yellow if the image storage is
almost full, red if the camera is disabled but needed for the run, grey
if the camera settings are is ready to be confirmed, and green if the
camera settings have been confirmed.

## Test Plan and Hands on Testing

- smoke tested on robot


https://github.com/user-attachments/assets/144dce45-58f7-4da8-b9d0-c7a819ca2176

## Changelog

- an additional param was added to check the storage info
- camera set up is now being viewed as a module set up type to allow for
checking of camera enablement and storage

## Review requests

- What do you need from reviewers to feel confident this PR is ready to
merge?
- Ask questions.

## Risk assessment
- lowish
- may have accidentally changed other text formatting

Closes
RQA-4828

ddcc4 committed:
Merge back 'chore_release-pd-8.6.3' into 'chore_release-8.8.0' (#20083)

Resolved conflicts:
- protocol-designer/release-notes.md

mjhuff committed:
fix(app): Fix "robot is busy" banners (#20088)

We render "robot is busy" banners on the robot details and robot settings views, however, the conditions in which we disable certain actions (ex., uploading zip files to a robot or allowing recalibration) are inconsistent from the conditions in which we render the banner.

Currently, we disable or enable certain actions based on a variety of factors:

* Whether the robot is "busy", which is roughly defined as whether there is a maintenance run or protocol run in progress, or there is an active subsystem update.
* Whether a protocol run exists.
* Whether a user has clicked specific module CTAs. The implementation of this one currently lacks a way to disable the "robot is busy" state, so once those specific items were clicked, the robot was busy so long as a user was on the robot settings page.

Another downstream consequence of this logic is that we permit desktop users to launch a maintenance run flow while another maintenance run is active.

The solution in this commit is the following:

* No actions of any kind should be permitted if the robot is busy (see the first bullet, above). This is the most conservative approach, but it is by far the least surprising to end-users. There's no more confusion why certain settings are disabled without the banner rendering and vice versa. For example, we could in theory update camera settings while a run is in progress, but under this proposal, a user wouldn't be permitted to do so. The additional (and more important benefit I think), is the code is significantly more auditable - we invoke the useIsRobotBusy hook exactly once at the top level component responsible for rendering all the settings, and that's it.
* We only block updating by zip file if there is a current, non-terminal run.

NOTE: The one spot that is quite annoying is still the whole "run uncurrenting" logic, which will cause the banner to persist when an end-user may think it shouldn't. This isn't a regression introduced by this commit, but yeah, if a user over-eagerly navigates away from a run and the ODD can't implicitly close the run context, then the banner persists...but that's a different problem for a different day I think.

ncdiehl11 committed:
fix(protocol-designer): filter incompatible lid parents in moveLabware (#20079)

Filter out incompatibble lid move locations in LabwareLocationField.
Potential labware or module move targets must be specified in the lid's
definition under `stackingOffsetWithLabware`,
`stackingOffsetWithModule`, or `comaptibleParentLabware` properties.

rclarke0 committed:
fix(shared-data): Add missing labware images and fix brand casing inconsistencies (#20093)

# Overview

Added missing labware images and fixed brand casing inconsistencies

## Test Plan and Hands on Testing

Smoke tested on labware-library
<img width="422" height="349" alt="Screenshot 2025-11-10 at 11 00 42 AM"
src="https://github.com/user-attachments/assets/10221ea4-8f57-4b76-9c6f-e265f1ee603c"
/>
<img width="424" height="351" alt="Screenshot 2025-11-10 at 11 00 56 AM"
src="https://github.com/user-attachments/assets/accb27e7-9879-4509-addd-0a259e8593a4"
/>
<img width="428" height="353" alt="Screenshot 2025-11-10 at 11 01 07 AM"
src="https://github.com/user-attachments/assets/85e50709-a7bb-48da-9f0c-94d702144494"
/>
<img width="429" height="349" alt="Screenshot 2025-11-10 at 11 02 40 AM"
src="https://github.com/user-attachments/assets/a6dd5ac9-e9a1-4611-8bb1-5f7f8ed1726c"
/>


## Risk assessment

low

Finishes AUTH-1941 & EXEC-1770
Other labware do not have tickets associated.

mjhuff committed:
feat(app): Add livestream duration event (#20094)

Closes EXEC-2049

mjhuff committed:
Merge branch 'chore_release-8.8.0' into 8.8.0-into-edge-incremental-mergeback

mjhuff committed:
Merge branch '8.8.0-into-edge-incremental-mergeback' into edge

ncdiehl11 committed:
feat(protocol-designer): handle full tip selection (#20101)

This PR adds the correct state for the SelectTips component for when all
pickup selections have been made. Specifically, it wires up the success
Chip component in the top right of the component above the deck map, and
adds an error state for all non-deselect hovers if there are no pickups
remaining.

rclarke0 committed:
fix(protocol-designer): correct camera step description (#20098)

# Overview

Correct camera step description

## Test Plan and Hands on Testing
<img width="267" height="767" alt="Screenshot 2025-11-10 at 12 48 56 PM"
src="https://github.com/user-attachments/assets/b5fb586f-8ffe-4033-8344-acf7d2e7b83e"
/>

Smoke tested

## Risk assessment

low

ncdiehl11 committed:
feat(protocol-designer, shared-data): refine OT-2 8-channel shadow and offset (#20106)

This PR adds formerly missing bounding box points for OT-2 GEN2
8-channel pipettes (P20, P300, and P1000). Also, it adds a custom
pipette shadow label offset for the specific case of OT-2 8-channel
pipettes, with a right label placement, to account for the irregular
shadow shape. Lastly, it removes erroneous well offsets for pipette
shadow placement (the well position (x, y) itself points to the center
of the well).

mjhuff committed:
refactor(mono, robot-server, app): Add notification support for `/camera` (#20105)

mjhuff committed:
refactor(app, components): enforce strict typing for icon data (#20112)

koji committed:
feat(app): add snack bar for non supported protocol (#20102)

* feat(app): add snack bar for non supported protocol

jerader committed:
refactor(app): delete the QT json ff completely (#20073)

closes AUTH-2473

ncdiehl11 committed:
fix(protocol-designer): allow return tip for single channel (#20116)

This PR unblocks single channel return tip

mjhuff reviewed (commented):
Agreed that this should be targeting `edge`, but I think this PR is pulling in commits from chore_release.

koji committed:
feat(app): support left col responsiveness (#20113)

* feat(app): support left col responsiveness

koji committed:
feat(app): make controls component responsive (#20115)

* feat(app): make controls component responsive

ddcc4 committed:
fix(components): don't forward our `showBorder` prop to <Box> and <div> (#20114)

# Overview

The console is cluttered up with warnings of:
```
Warning: React does not recognize the `showBorder` prop on a DOM element.
If you intentionally want it to appear in the DOM as a custom attribute,
spell it as lowercase `showborder` instead. If you accidentally passed it
from a parent component, remove it from the DOM element.
```

`showBorder` (and `isZoomed`) is a custom property we invented for
`<StyledBox>`. We need to prevent it from being forwarded down to the
`<div>`.

## Test Plan and Hands on Testing

Tried it in PD running locally.

## Review requests

I haven't touched styled components before, so I don't really know what
I'm doing.

Also, it seems like a lot of the existing code says:
```
{ shouldForwardProp: isntStyleProp }
```
Isn't that the opposite of what we want? This is saying that if
**isn't** one of the native style properties, then we **do** forward it.
Don't we want this to be `{ shouldForwardProp: isStyleProp }` instead?

## Risk assessment

Low.

rclarke0 committed:
Merge branch 'EXEC-2043' of https://github.com/Opentrons/opentrons into EXEC-2043
